packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -278,7 +278,7 @@ class IOSDevice extends Device {
 
       Future<int> launch = runCommandAndStreamOutput(launchCommand, trace: true);
 
-      List<Uri> uris = await launch.then((int result) async {
+      List<Uri> uris = await launch.then<List<Uri>>((int result) async {
```

Comment:
It was definitely failing `flutter analyze` w/ `dev.10.1` , though interestingly I was not seeing the errors in IntelliJ.

Maybe @danrubel can comment on that disconnect.  I haven't had time to follow-up on it...